uint16[7] enc_position
uint16 gripper_data